CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -196,12 +202,15 @@ void full_colour_renderer::decode_image(const uint8_t* pImg,
       case 3:  // Recolour layer
       {
         uint8_t iTable = *pImg++;
+        uint32_t iColour;
         pImg++;  // Skip reading the opacity for now.
         if (iTable == 0xFF) {
           // Legacy sprite data. Use the palette to recolour the
           // layer. Note that the iOpacity is ignored here.
           while (iLength > 0) {
-            push_pixel(pColours[*pImg++]);
+            iColour = pColours[*pImg++];
+
+            push_pixel(iColour);
```

Comment:
This looks like a unneeded change, code hasn't changed functionally, and you don't use the new variable elsewhere.

Reply:
if teh new variable is required, at least make it local to its use.